### DISCUSSION TITLE
Your Ruby version is 2.6.8, but your Gemfile specified 2.7.6 error while enabling new architecture on react-native iOS

### DISCUSSION THREAD

**Discussion by BraveEvidence** [reactwg/react-native-new-architecture#108] [Q&A] [answered]
Sorry I have already asked the same question here but this time even after following all the steps I get the error

I have already gone through [this](https://stackoverflow.com/questions/74189532/your-ruby-version-is-2-6-8-but-your-gemfile-specified-2-7-5) post

I am facing this issue on my new react-native project version 0.71.1 while enabling the new architecture for iOS

I have done all the below steps

```
$ brew update
$ brew install ruby-build
$ brew install rbenv

$ rbenv install 2.7.6
$ rbenv global 2.7.6

echo 'export PATH="$HOME/.rbenv/bin:$PATH"' >> ~/.zshrc
echo 'eval "$(rbenv init -)"' >> ~/.zshrc
```

When I run `ruby --version` I get ruby `2.7.6p219 (2022-04-12 revision c9c2245c0a) [x86_64-darwin22]` which is the correct ruby version required to enable new architecture on iOS

but when I run `bundle install && RCT_NEW_ARCH_ENABLED=1 bundle exec pod install` I still get error saying

Your Ruby version is 2.6.10, but your Gemfile specified 2.7.6

<img width="1155" alt="MJ3yp" src="https://user-images.githubusercontent.com/24708929/215389622-941eb7cf-27c9-4f94-ac8b-b84f1cf88dcf.png">
``

**Comment by cipolleschi** [comment, marked as answer]
You also have to kill the terminal and reopen it, otherwise the `eval "$(rbenv init -)"' >> ~/.zshrc` is not actually run.

Basically, you did all the steps properly, but the Terminal is not updated to see the new versions.

**Reply by BraveEvidence** [reply]
Already did it multiple times but still same issue

**Reply by cipolleschi** [reply]
Please, avoid spamming the same error in different places. You posted the same problem also [here](https://github.com/facebook/react-native/issues/36002). Let's continue the discussion there.

**Reply by leotm** [reply]
if it helps
- https://github.com/facebook/react-native/issues/35807#issuecomment-1378831502

**Comment by BraveEvidence** [comment]
I was able to resolve my issue by running `bundle install --redownload`. It Forces a redownload of all gems on the gemfile, assigning them to the new bundler
